src/domain/core/src/use_cases/create_dataset_use_case.rs

```diff
@@ -15,6 +15,10 @@ use crate::{CreateDatasetError, CreateDatasetResult};
 
 #[async_trait::async_trait]
 pub trait CreateDatasetUseCase: Send + Sync {
+    // TODO: Add "options" that specify the visibility dataset
+    //
+    //       Based on: Private Datasets: Update use case: Pushing a dataset
+    //                 https://github.com/kamu-data/kamu-cli/issues/728
```

Comment:
Why is this different from snapshot? I think you should share a common structure

Reply:
Yeah, there's no point in putting it off until this ticket

Done